src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -1422,4 +1463,22 @@ private void updateTexts(UndoChangeEvent event) {
             */
         }
     }
+
+    private void copyGroupTreeNode(LibraryTab libraryTabCasted, Optional<GroupTreeNode> parent, Optional<GroupTreeNode> groupTreeNodeToCopy, List<BibEntry> allEntries) {
+        // add groupTreeNodeToCopy to the parent-- in the first run that will the source/main GroupTreeNode
+        GroupTreeNode copiedNode = parent.get().addSubgroup(groupTreeNodeToCopy.get().copyNode().getGroup());
+
+        // add all entries of a groupTreeNode to the new library.
+        libraryTabCasted.dropEntry(groupTreeNodeToCopy.get().getEntriesInGroup(allEntries));
+
+        // List of all children of groupTreeNodeToCopy
+        List<GroupTreeNode> children = groupTreeNodeToCopy.get().getChildren();
+
+        if (!children.isEmpty()) {
+            // use recursion to add all subgroups of the original groupTreeNodeToCopy
+            for (GroupTreeNode child : children) {
+                copyGroupTreeNode(libraryTabCasted, Optional.of(copiedNode), Optional.of(child), allEntries);
```

Comment:
An alternative way to copying groups https://github.com/JabRef/jabref/blob/e86839a85fb939920e3a3d0214aeda9a0b47ae44/src/main/java/org/jabref/gui/collab/groupchange/GroupChange.java#L44-L46